lms/sql_tasks/tasks/report/create_from_scratch/02_entities/05_events/01_create_view.sql

```diff
@@ -10,32 +10,32 @@ DROP MATERIALIZED VIEW IF EXISTS report.events CASCADE;
 
 CREATE MATERIALIZED VIEW report.events AS (
     WITH
+        -- Unique event user relations without considering role
+        unique_event_users AS (
+            SELECT DISTINCT event_id, user_id FROM event_user
+        ),
+
         translated_events AS (
             SELECT
                 DATE_TRUNC('week', timestamp)::date AS timestamp_week,
                 user_map.organization_id,
                 event_type.type::report.event_type AS event_type,
                 user_map.user_id
             FROM event
-            JOIN event_user ON
-                event_user.event_id = event.id
             JOIN event_type ON
                 event.type_id = event_type.id
-            JOIN report.user_map ON
-                event_user.user_id = user_map.lms_user_id
-            GROUP BY
-                timestamp, event_id, event_type.type, user_map.user_id, user_map.organization_id
+            -- Lots of events don't have users
+            LEFT OUTER JOIN unique_event_users ON
+                unique_event_users.event_id = event.id
+            LEFT OUTER JOIN report.user_map ON
+                unique_event_users.user_id = user_map.lms_user_id
```

Comment:
This is the meat of the change here, making these left outers, so that rows without users still show up

Reply:
Yeah, missed this too, for launches missing users it's about historical data being incomplete but for other types of events, it might not make sense to link them to a particular user.